unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1726,6 +1681,85 @@ handleCreatePullRequest baseRepo0 headRepo0 = do
               diff <- mergeAndDiff baseBranch headBranch
               respondNumbered diff
 
+handleFindI
+  :: (Monad m, Var v)
+  => Bool
+  -> FindScope
+  -> [String]
+  -> Input
+  -> Action' m v ()
+handleFindI isVerbose fscope ws input = do
+  root' <- use LoopState.root
+  currentPath' <- use LoopState.currentPath
+  currentBranch' <- getAt currentPath'
+  let currentBranch0 = Branch.head currentBranch'
+      getNames :: FindScope -> Names
+      getNames findScope =
+        let namesWithinCurrentPath = Backend.prettyNamesForBranch root' nameScope
```

Comment:
We should really be caching `currentBranchNames` in the Action monad somewhere 😅 ; probably not worth changing here though.